src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx

```diff
@@ -229,6 +247,25 @@ describe('<BudgetDetailPage />', () => {
       userId: 3,
     });
 
+    // Mock useBudgetId to be dynamic based on useParams
+    useBudgetId.mockImplementation(() => {
+      const { budgetId } = useParams();
+      const isUUID = (id) => /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/.test(id);
```

Comment:
[nit] the `uuid` package has a function to validate if a value is a true uuid if we want to avoid regex.
https://github.com/uuidjs/uuid?tab=readme-ov-file#uuidvalidatestr